grails-app/controllers/org/pih/warehouse/shipping/ShipmentController.groovy

```diff
@@ -1189,6 +1192,33 @@ class ShipmentController {
         redirect(controller: "shipment", action: "showDetails", id: params.id)
     }
 
+    def downloadInvoiceTemplateDocument = {
+        Shipment shipmentInstance = Shipment.get(params.shipment.id)
+
+        if (!shipmentInstance) {
+            throw new Exception("Unable to locate shipment with ID ${params.id}")
+        }
+
+        ShipmentWorkflow shipmentWorkflow = shipmentService.getShipmentWorkflow(shipmentInstance)
+        def documentInstance = shipmentWorkflow.documentTemplates?.find {it.documentType?.documentCode == DocumentCode.INVOICE_TEMPLATE}
+
+        if (!documentInstance) {
+            throw new Exception("Unable to locate document with type ${DocumentCode.INVOICE_TEMPLATE}. Please add ${DocumentCode.INVOICE_TEMPLATE} document to shipment workflow.")
+        } else if (documentInstance && shipmentInstance) {
```

Comment:
`else` is not needed here, there will be an exception thrown if any of this is false